### PR TITLE
RTL Bug fix

### PR DESCRIPTION
**Pull request by hosseinkhaledi**
1- Add rtl-prop to use `@include rtl-prop(margin-left, margin-right, 10);` instead of `rtl(margin-left, 10, inherent), rtl(margin-right, inherent, 10)`

2- Change rtl function to apply ltr rules only in ltr document
3- Remove some component bug in rtl mode

### CODE REVIEW

File: src/components/list/list.scss [hunk 1]
```diff
@@ -143,18 +143,18 @@ md-list-item, md-list-item .md-list-item-inner {
...
   & > md-icon:first-child:not(.md-avatar-icon) {
-    margin-right: $list-item-primary-width - $list-item-primary-icon-width;
+    @include rtl-prop(margin-right, margin-left, $list-item-primary-width - $list-item-primary-icon-width);
   }
   & > md-checkbox {
     width: 3 * $baseline-grid;
-    margin-left: 3px;
-    margin-right: 29px;
+    @include rtl(margin-left, 3px, 29px);
+    @include rtl(margin-right, 29px,3px);
```
Comment (DevVersion) [inline src/components/list/list.scss]: Little careless style mistake here?

**Comment by DevVersion** [comment]
I think this are good changes for the actual RTL support. May you review this @EladBezalel?

**Comment by EladBezalel** [comment]
This is a blessed contribution, remembered your name from last pr (#4611) will probably fix #1080.

File: src/components/list/list.scss (hunk 1 above)
Comment (hosseinkhaledi) [inline src/components/list/list.scss]: no i check and it is ok
in ltr mode is
    margin-left: 3px;
    margin-right: 29px; 
in rtl mode is
   margin-left: 29px
   margin-right: 3px

Comment (DevVersion) [inline src/components/list/list.scss]: Not meant that - meant the actual space between the parameters and their comma. `,[]3px`, but as I said, it's a careless style mistake.

Comment (hosseinkhaledi) [inline src/components/list/list.scss]: i dont know that, thank u

**Comment by ThomasBurleson** [comment]
@hosseinkhaledi - Kudos!  Thank you.

**Comment by ThomasBurleson** [comment]
Closed with SHA 08bf4f93763d2aea9901d697f8cc3786d242725f

**Pull request closed without merging**